utils/analysis/helpers.go

```diff
@@ -87,7 +87,9 @@ func IsTerminating(run *v1alpha1.AnalysisRun) bool {
 	for _, res := range run.Status.MetricResults {
 		switch res.Phase {
 		case v1alpha1.AnalysisPhaseFailed, v1alpha1.AnalysisPhaseError, v1alpha1.AnalysisPhaseInconclusive:
-			return true
+			// If this metric is running in the dryRun mode then we don't care about the failures and hence the terminal
+			// decision shouldn't be affected.
+			return !res.DryRun
```

Comment:
so if analysisRun is completed, then dryRun will keep it in running mode (argoCD UI shows as spinning and that will confuse users).
Don't we need default state or ignore analysis run state when dryRun is enabled

Reply:
maybe I am wrong here.. eventually, analysis run will be completed

Reply:
@harikrongali Yes, it'll only keep running if it's a background analysis where the `count` is not specified. Think of it as the case where there are no failures/errors and the AR keeps running in background until the rollout finishes.